### PR TITLE
#162949035 Users should be able Signup/Register to Authors Haven

### PR DESCRIPTION
**Pull request by johnwayodi**
#### What does this PR do?

Creates user Interfaces through which a user can be able to register and verify an account in the application.

#### Description of Task to be completed?

- [x] Create a folder structure
- [x] Create React components, views
- [x] Create React Views
- [x] Design the Register User Interface
- [x] Design Verification User Interface
- [x] Configure Redux and saga middleware
- [x] Consume the Register API
- [x] Consume Verification API
- [x] Add Tests

#### How should this be manually tested?

1. Clone this repository on your local machine using 

     `git clone https://github.com/andela/ah-robotics-frontend.git`
2. Change your working directory to the project's directory using 
     
     `cd ah-robotics-frontend`
3. Change to the specific branch `ft-user-register-162949035` using 

     `git checkout ft-user-register-162949035`

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm test` command to ensure the tests are passing.

    #### Testing the features
6. On the browser, head over to the URL 
`https://ah-robotics-frontend-sta-pr-12.herokuapp.com/register`, 
 The register page should be displayed as shown below.

screen shot 2019-02-18 at 7 42 28 am

7. Enter the various details and click on **Register** button to create a new user. A verification email containing a link will be sent to the email provided during registration, As shown below:

screen shot 2019-02-18 at 7 51 54 am

8. Upon clicking the link, the user will be redirected to a page displaying successful verification.

screen shot 2019-02-18 at 7 44 53 am

#### What are the relevant pivotal tracker stories?

[#162949035](https://www.pivotaltracker.com/n/projects/2235171/stories/162949035)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,12 +15,12 @@ const App = () => (
       <Switch>
         <Route path="/" component={HomeView} exact />
         <Route path="/login" component={LoginView} exact />
+        <Route path="/register" component={RegisterView} />
+        <Route path="/verify/:token" component={VerifyRegistrationView} />
         <Route component={Error} />
-
```
Comment (jomasim) [inline src/App.js:17]: remove the spaces

File: src/redux/actions/RegisterActions/actions.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    // const text = 'Finish docs'
+    const expectedAction = {
+      type: types.REGISTER_START,
+      payload: user,
+    };
+    expect(actions.registerUser(user)).toEqual(expectedAction);
+  });
+
+  it('should create an action to indicate successful registration', () => {
+    // const text = 'Finish docs'
```
Comment (jomasim) [inline src/redux/actions/RegisterActions/actions.test.js]: remove the commented code

**Review by jomasim — CHANGES_REQUESTED** [review]

File: src/redux/actions/RegisterActions/actions.test.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    const expectedAction = {
+      type: types.REGISTER_SUCCESS,
+      payload: user,
+    };
+    expect(actions.registerSuccess(user)).toEqual(expectedAction);
+  });
+
+  it('should create an action to indicate failed registration', () => {
+    // const text = 'Finish docs'
```
Comment (jomasim) [inline src/redux/actions/RegisterActions/actions.test.js]: remove the commented code

**Review by jomasim — CHANGES_REQUESTED** [review]

**Comment by jomasim** [comment]
@johnwayodi confirm password is not working. The signup works even with two unmatching passwords.

**Comment by jomasim** [comment]
@johnwayodi  @waracci What message does user expect upon a successful verification? What I have here is page does not exist.
screenshot 2019-02-18 at 14 25 03

@johnwayodi  @waracci It works now 👍 
screenshot 2019-02-18 at 14 41 21

**Review by jomasim — APPROVED** [review]

File: src/redux/actions/RegisterActions/actions.test.js (hunk 3 above)
Comment (johnwayodi) [reply src/redux/actions/RegisterActions/actions.test.js]: @jomasim we have removed all the commented code. Thanks for reviewing.

**Comment by Teatoller** [comment]
Hello,

I have managed to test the branch successfully.

Regards
Steven Ennis

**Review by sharkdevs — COMMENTED** [review]
It is working fine, Been able to get verified.
screen shot 2019-02-18 at 14 50 05

**Review by Paulvitalis200 — COMMENTED** [review]
Hey @wayodi and @warraci,
Hey I managed to test your Pull Request and everything works well.

**Review by Collins33 — APPROVED** [review]

**Pull request merged**